sdk/typescript/src/transactions/executor/parallel.ts

```diff
@@ -197,6 +240,8 @@ export class ParallelTransactionExecutor {
 				}
 			}
 
+			this.#lastDigest = results.digest;
```

Comment:
The executor tracks a list of gas coins to use to refill the pool, this contains the coins from failed executions and coins who's balance was too low.  we don't track the versions for these coins, because in error cases, they might become invalid.  By tracking the digest of the last executed transaction, we can wait for that digest to be indexed before executing the refill transaction, which should ensure the index is up-to-date for all coins used to refill the pool.